### PR TITLE
perf(radial): vectorize get_radial_zernikes via shared _zernike_scores (~2x)

### PR DESCRIPTION
**Pull request by timtreis**
## What

`get_radial_zernikes` already built the Zernike basis on the masked foreground vectors (cheap), but reduced it with **2·K separate `scipy.ndimage.sum_labels` calls** (one per moment × real/imag) and **re-gathered `pixels[ijv]` inside every one**. Profiling showed the reduction — not the basis — dominated.

This delegates the intensity-weighted moment sums to the shared **`cp_measure.utils._zernike_scores`** (added in #74 for `get_zernike`) with the pixel image as the per-pixel `weight`. The helper keeps the basis on the foreground vectors and segment-sums each moment by label with a single `numpy.bincount`, collapsing the whole reduction into one vectorised pass. The caller then normalises by each object's **pixel count** (radial Zernikes divide by pixel count, not the enclosing-circle area) and forms magnitude/phase.

This cashes in the `weight=` reuse hook deliberately left on `_zernike_scores`.

### CODE REVIEW

File: src/cp_measure/core/measureobjectintensitydistribution.py [hunk 1]
```diff
@@ -317,73 +317,32 @@ def get_radial_zernikes(
...
-
-    unique_labels = numpy.arange(1, int(labels.max()) + 1)
-    # MODIFIED: Delegate index generation to the minimum_enclosing_circle
-    # MODIFIED: We assume non-overlapping labels for now
-    # TODO Support label overlap (i.e., format in ijv)
-    # MODIFIED: Delegate indexes to minimum_enclosing_circle
-    ij, r = centrosome.cpmorphology.minimum_enclosing_circle(labels, unique_labels)
+    zernike_indexes = zernike.get_zernike_indexes(zernike_degree + 1)
+
+    # The intensity-weighted moment sums are exactly `_zernike_scores` with the pixel image
```
Comment (afermg) [inline src/cp_measure/core/measureobjectintensitydistribution.py]: As I said in the chat, this is a lot in comments to take in. I think we should instead have a very short version of this and reconstruct this explanation as the docstring over the _zernike_scores definition.

File: src/cp_measure/core/measureobjectintensitydistribution.py [hunk 2]
```diff
@@ -317,73 +317,32 @@ def get_radial_zernikes(
...
+    # as the per-pixel weight: vr[obj, k] = sum_pixels intensity * Re(basis_k), and likewise
+    # for the imaginary part. The shared helper keeps the basis on the foreground vectors (no
+    # full (H, W, K) scatter, no 2*K `scipy.ndimage.sum_labels` passes) and returns each
+    # object's pixel `counts` — radial Zernikes normalise by pixel count, not the
+    # enclosing-circle area. Labels are the contiguous 1..N cp_measure guarantees (see
+    # cp_measure._sanitize). With no objects it returns empty (0, K) arrays, so the loop below
+    # yields empty features without a special case.
+    vr, vi, _radii, counts = _zernike_scores(labels, zernike_indexes, weight=pixels)
```
Comment (afermg) [inline src/cp_measure/core/measureobjectintensitydistribution.py:324]: It is great how the complexity is abstracted away (since the zernike scoring is used in two different measurements)

File: test/test_radial_zernike.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+"""Golden + edge tests for the vectorised numpy ``get_radial_zernikes``.
+
+The rewrite delegates the intensity-weighted moment sums to the shared
```
Comment (afermg) [inline test/test_radial_zernike.py]: This doesn't seem relevant for the tests

Comment (afermg) [reply test/test_radial_zernike.py]: We should have "what behaviours are we testing?" Based on the code it seems to be the numerical fidelity relative to the original implementation (instead of the numerical results of the reference data).

File: test/test_radial_zernike.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    M_CATEGORY,
+    get_radial_zernikes,
+)
+from cp_measure.utils import masks_to_ijv
+
+ATOL = 1e-9  # >> the ~1e-13 summation-order round-off, << any real signal
+
+
+def _reference(labels, pixels, zernike_degree=9):
+    """Direct centrosome path, called with the contiguous 1..N label values."""
```
Comment (afermg) [inline test/test_radial_zernike.py]: Worth documenting in the docstring that this follows the original implementation.

File: test/test_radial_zernike.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    # sweeping the degree checks the basis truncation.
+    rng = numpy.random.default_rng(0)
+    masks = numpy.zeros((128, 128), numpy.int32)
+    for lab, (cy, cx) in enumerate(rng.integers(20, 108, size=(6, 2)), 1):
+        yy, xx = numpy.mgrid[0:128, 0:128]
+        masks[(yy - cy) ** 2 + (xx - cx) ** 2 < rng.integers(40, 120)] = lab
+    _assert_matches(masks, _pixels(masks.shape), zernike_degree)
+
+
+def test_radial_zernike_object_touching_edge():
```
Comment (afermg) [inline test/test_radial_zernike.py:88]: Is there any particular reason why we ought to test objects touching the edge?

File: test/test_radial_zernike.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    # minimum_enclosing_circle radius -> 0, so both paths divide by zero identically.
+    masks = numpy.zeros((32, 32), numpy.int32)
+    masks[16, 16] = 1
+    masks[5:15, 5:15] = 2  # a normal object alongside the degenerate one
+    _assert_matches(masks, _pixels(masks.shape))
+
+
+def test_radial_zernike_empty_mask():
+    masks = numpy.zeros((40, 40), numpy.int32)
+    got = get_radial_zernikes(masks, _pixels(masks.shape))
```
Comment (afermg) [inline test/test_radial_zernike.py]: I feel like the duplicated logic here can be avoided if _assert_matches supports the empty case. Please consider adding support for that there (it's a very minor detail though).

**Review by afermg — APPROVED** [review]
Looks good, I suggested minor changes (mostly about testing, since most changes already occurred in a previous PR). I will approve preemptively. If you merge it yourself after those minor changes please squash.

File: test/test_radial_zernike.py (hunk 5 above)
Comment (timtreis) [reply test/test_radial_zernike.py:88]: It's a weird edge case that I'd rather not be surprised by when we speed up things 🤔
